src/components/PostItem.tsx

```diff
@@ -0,0 +1,55 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+
+import classNames from 'classnames';
+import React from 'react';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { selectPost } from '../features/selectedPost';
+
+type Props = {};
+
+export const PostsList: React.FC<Props> = ({}) => {
+  const dispatch = useAppDispatch();
+  const { posts } = useAppSelector(state => state.posts);
+  const selectedPost = useAppSelector(state => state.selectedPost);
+
+  return (
+    <div data-cy="PostsList">
+      <p className="title">Posts:</p>
+
+      <table className="table is-fullwidth is-striped is-hoverable is-narrow">
+        <thead>
+          <tr className="has-background-link-light">
+            <th>#</th>
+            <th>Title</th>
+            <th> </th>
+          </tr>
+        </thead>
+
+        <tbody>
+          {posts.map(post => (
+            <tr key={post.id} data-cy="Post">
+              <td data-cy="PostId">{post.id}</td>
+              <td data-cy="PostTitle">{post.title}</td>
+              <td className="has-text-right is-vcentered">
+                <button
+                  type="button"
+                  data-cy="PostButton"
+                  className={classNames('button', 'is-link', {
+                    'is-light': post.id !== selectedPost?.id,
+                  })}
+                  onClick={() => {
+                    dispatch(
+                      selectPost(post.id === selectedPost?.id ? null : post),
+                    );
+                  }}
+                >
+                  {post.id === selectedPost?.id ? 'Close' : 'Open'}
+                </button>
+              </td>
+            </tr>
+          ))}
```

Comment:
Use desctructure syntax whenever you can

Reply:
I need to use the whole obj 'post' in map method to get access to onClick method. In case of destructure id and title, i wont be able to get scope.

Reply:
Oh okay. I didn;t notice that